syntax = "proto3";
package me.sylver.marver.sentry;

service SentryService {
    rpc GetVector (GetVectorRequest) returns (GetVectorResponse) {}
    rpc DetectFaces (DetectFacesRequest) returns (DetectFacesResponse) {}
}

message GetVectorRequest {
    oneof input {
        string file_path = 4;
        string text_input = 5;
    }
}

message GetVectorResponse {
    Vector vector = 1;
}

message DetectFacesRequest {
    string file_path = 1;
}

message DetectFacesResponse {
    repeated Face faces = 1;
}

message Face {
    BoundingBox bounding_box = 1;
    Vector vector = 2;
    float confidence = 3;
}

message BoundingBox {
    float x1 = 1;
    float y1 = 2;
    float x2 = 3;
    float y2 = 4;
}

message Vector {
    repeated float value = 1;
}
